scene/main/viewport.cpp

```diff
@@ -738,6 +738,14 @@ void Viewport::_process_picking() {
 
 	while (physics_picking_events.size()) {
 		local_input_handled = false;
+		if (!handle_input_locally) {
+			Viewport *vp = this;
+			while (!Object::cast_to<Window>(vp) && vp->get_parent()) {
```

Comment:
Assuming it's guaranteed this will always be non-null?

Reply:
Physics picking happens only when the viewport is inside the scene-tree:

https://github.com/godotengine/godot/blob/74c32faa78b54863f8f25c538083907c2bf71791/scene/main/viewport.cpp#L676-L679

So the returned viewport is also inside the scene-tree.

The same method is used in preexisting code:

https://github.com/godotengine/godot/blame/d76c1d0e516fedc535a2e394ab780cac79203477/scene/main/viewport.cpp#L3657-L3673